crates/noirc_abi/src/input_parser/toml.rs

```diff
@@ -4,9 +4,11 @@ use std::{collections::BTreeMap, path::Path};
 
 use super::InputValue;
 
-pub(crate) fn parse<P: AsRef<Path>>(path_to_toml: P) -> BTreeMap<String, InputValue> {
+pub(crate) fn parse<P: AsRef<Path>>(path_to_toml: P) -> Result<BTreeMap<String, InputValue>, String> {
```

Comment:
Instead of a string error type, can we use a dedicated error enum?

Reply:
I was using String as these errors are only used right now to construct a `CliError` that lives inside the `nargo` crate. Without moving the `CliError`, using it in the `noirc_abi` would also cause a cyclic import so I opted for string for now. I could add a separate error enum for the types of errors in the `input_parser`. I was unsure if I should originally as the error would have to be deconstructed right away in `prove_cmd.rs` and `verify_cmd.rs`. 

`CliError` could be moved though to `noirc_abi/input_parser`, but with these names it felt a little out of place to me. `CliError` though is used across multiple CLI commands for problems all relating to the running of a Noir program. I could move move the error enum to `input_parser` and rename it to something like `ProgramInputError` (or keep the name as). What do you think?

Reply:
I was thinking a new error enum only encapsulating these new errors could be used. Along with a From impl to convert to CliError via `?` perhaps. If that doesn't work or is too much work then I'm fine with keeping it a String here.